module/zfs/vdev.c

```diff
@@ -2759,23 +2759,24 @@ vdev_resilver_needed(vdev_t *vd, uint64_t *minp, uint64_t *maxp)
 
 /*
  * Gets the checkpoint space map object from the vdev's ZAP.
- * Returns the spacemap object, or 0 if it wasn't in the ZAP
- * or the ZAP doesn't exist yet.
+ * Returns the spacemap object, or 0 if it wasn't in the ZAP,
+ * the ZAP doesn't exist yet, or the ZAP is damaged.
  */
 int
 vdev_checkpoint_sm_object(vdev_t *vd)
 {
+	int err __attribute__((unused));
+
 	ASSERT0(spa_config_held(vd->vdev_spa, SCL_ALL, RW_WRITER));
 	if (vd->vdev_top_zap == 0) {
 		return (0);
 	}
 
 	uint64_t sm_obj = 0;
-	int err = zap_lookup(spa_meta_objset(vd->vdev_spa), vd->vdev_top_zap,
+	err = zap_lookup(spa_meta_objset(vd->vdev_spa), vd->vdev_top_zap,
```

Comment:
Can we maybe say `int err = .....` here and then later have something like this:
```
if (err != 0 && err != ENOENT) {
    ASSERT3S(err, ==, ECKSUM);
    vdev_dbgmsg(vd, "vdev_load: vdev_checkpoint_sm_objset "
        "failed to retrieve checkpoint space map object from vdev ZAP "
        "[error=%d]", error);
}
```

If something that bad happens we would want to at least log it somewhere.